src/cli/help.rs

```diff
@@ -1,4 +1,4 @@
-pub static RUSTUP_HELP: &'static str = r"DISCUSSION:
+pub const RUSTUP_HELP: &str = r"DISCUSSION:
```

Comment:
I don't think we should make this change - we might want to compare string pointers at some point and this makes that a footgun (or hopefully impossible?). In general, I think use consts when we care about the value, and statics where we care about the pointer (which I think we do for strings, etc).

Reply:
This const is a const to a pointer I think.

Reply:
So the `const` keyword differs with `static` that it makes this string
is immutable. Everything else is the same between these keywords.

Reply:
While the address of a `const` is is not fixed, if the constant is itself a pointer then *that* pointer **is** fixed.

Reply:
I think (I'm actually not sure about this) that the compiler can still duplicate the data with consts. I'm not sure if that is better or worse

Reply:
I built rustup in release mode, the binary size of with this PR is the same
with master branch.

Reply:
After some more investigation, the address of a constant, or of any pointers in that constant, is not fixed. This can be easily demonstrated by having two dylibs use the same constant string and print different addresses for the string. The address of a static is fixed however. So if fixed addressing is *required* then use `static`, otherwise use `const`.

Reply:
I really don't get the point why we need fixed addresses of strings in binary file.

Reply:
At this point, if there's concern from at least one dev, why insist on this change blocking the rest of your cleanups?  It would make sense to me to remove this static->const change for now, to allow us a better chance to merge the rest (once the argument about `local` is dealt with)

Reply:
After read many comparisons between using const and static, I'm convinced 
that static would be better in this case.

Relevant links:
* https://doc.rust-lang.org/1.30.0/book/first-edition/const-and-static.html
* https://stackoverflow.com/questions/37877381/what-is-the-difference-between-immutable-and-const-variables-in-rust
* https://github.com/frol/rust-rfcs/blob/master/text/0246-const-vs-static.md

I will mark this conversation as resolved.